jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java

```diff
@@ -1,22 +1,61 @@
 package pl.droidsonroids.jspoon;
 
+import java.lang.reflect.Constructor;
 import java.math.BigDecimal;
 import java.util.Date;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+
 import org.jsoup.nodes.Element;
 
+import pl.droidsonroids.jspoon.exception.ConstructorNotFoundException;
+import pl.droidsonroids.jspoon.exception.ObjectCreationException;
+
 class Utils {
-    static boolean isSimple(Class clazz) {
-        return clazz.equals(String.class) ||
-                clazz.equals(Integer.class) || clazz.equals(int.class) ||
-                clazz.equals(Long.class) || clazz.equals(long.class) ||
-                clazz.equals(Float.class) || clazz.equals(float.class) ||
-                clazz.equals(Double.class) || clazz.equals(double.class) ||
-                clazz.equals(Boolean.class) || clazz.equals(boolean.class) ||
-                clazz.equals(BigDecimal.class) ||
-                clazz.equals(Element.class) ||
-                clazz.equals(List.class) ||
-                clazz.equals(Date.class) ||
-                clazz.equals(Element.class);
+
+    public final static Map<Class<?>, Class<?>> PRIMITIVE_WRAPPERS = new HashMap<Class<?>, Class<?>>();
+
+    static {
+        PRIMITIVE_WRAPPERS.put(boolean.class, Boolean.class);
+        PRIMITIVE_WRAPPERS.put(byte.class, Byte.class);
+        PRIMITIVE_WRAPPERS.put(short.class, Short.class);
+        PRIMITIVE_WRAPPERS.put(char.class, Character.class);
+        PRIMITIVE_WRAPPERS.put(int.class, Integer.class);
+        PRIMITIVE_WRAPPERS.put(long.class, Long.class);
+        PRIMITIVE_WRAPPERS.put(float.class, Float.class);
+        PRIMITIVE_WRAPPERS.put(double.class, Double.class);
+    }
+
+    public static <T> Class<T> wrapToObject(Class<T> clazz){
+        if (!clazz.isPrimitive()) {
+            return clazz;
+        }
+        @SuppressWarnings("unchecked")
+        Class<T> wrapped = (Class<T>) PRIMITIVE_WRAPPERS.get(clazz);
+        return (wrapped != null) ? wrapped : clazz;
+    }
+
+
+    public static <T> T constructInstance(Class<T> clazz) {
+        try {
+            Constructor<T> constructor = clazz.getDeclaredConstructor();
+            constructor.setAccessible(true);
+            return constructor.newInstance();
+        } catch (NoSuchMethodException e) {
+            throw new ConstructorNotFoundException(clazz.getSimpleName());
+        } catch (Exception e) {
+            throw new ObjectCreationException(clazz.getSimpleName());
```

Comment:
I think it is a good idea to set original exception `e` as a cause of `ObjectCreationException`. It can help with debugging.

Reply:
I agree. In fact I would like to enhance the whole exception herarchy.

Current exceptions only act as message decorators for an underlying RubtimeException. It would be great to have a possibility to differantiate between implementation exceptions such as above and parser exceptions. Also possibility to catch them and retrieve useful information examp. about field or node content, also underlying exception.
But I'll try to address all these in a separate PR